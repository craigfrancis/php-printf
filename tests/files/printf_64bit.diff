     *** Output for zero argument ***
002- printf() expects at least 1 argument, 0 given
002+ Too few arguments to function user_printf(), 0 passed in /Volumes/WebServer/Projects/php.fprint/tests/files/printf_64bit.php on line 21 and exactly 1 expected
     
     *** Output for insufficient number of arguments ***
005- Error found: 5 arguments are required, 3 given
006- *** Output for scalar argument ***
007- 3
     
009- *** Output for float type ***
006+ Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29
     
011-  Input Float numbers variation array is:
012- Array
013- (
014-     [0] => 0
015-     [1] => 1
016-     [2] => -1
017-     [3] => 0.32
018-     [4] => -0.32
019-     [5] => 3.4-3.4
020-     [6] => 2.54
021-     [7] => -2.54
022-     [8] => 1.2345678E+99
023-     [9] => -1.2345678E+99
024- )
025- 
026- 
027- Float Iteration 1
028- 0.000000
029- 1.000000
030- -1.000000
031- 0.320000
032- -0.320000
033- 3.400000
034- 2.540000
035- -2.540000
036- 1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
037- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
038- 
039- Float Iteration 2
040- 0.000000
041- 1.000000
042- -1.000000
043- 0.320000
044- -0.320000
045- 3.400000
046- 2.540000
047- -2.540000
048- 1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
049- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
050- 
051- Float Iteration 3
052- +0.000000
053- +1.000000
054- -1.000000
055- +0.320000
056- -0.320000
057- +3.400000
058- +2.540000
059- -2.540000
060- +1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
061- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
062- 
063- Float Iteration 4
064-    0.00
065-    1.00
066-   -1.00
067-    0.32
068-   -0.32
069-    3.40
070-    2.54
071-   -2.54
072- 1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
073- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
074- 
075- Float Iteration 5
076- 0.00   
077- 1.00   
078- -1.00  
079- 0.32   
080- -0.32  
081- 3.40   
082- 2.54   
083- -2.54  
084- 1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
085- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
086- 
087- Float Iteration 6
088- 0000.00
089- 0001.00
090- -001.00
091- 0000.32
092- -000.32
093- 0003.40
094- 0002.54
095- -002.54
096- 1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
097- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
098- 
099- Float Iteration 7
100- 0.00000
101- 1.00000
102- -1.0000
103- 0.32000
104- -0.3200
105- 3.40000
106- 2.54000
107- -2.5400
108- 1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
109- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
110- 
111- Float Iteration 8
112- ###0.00
113- ###1.00
114- ##-1.00
115- ###0.32
116- ##-0.32
117- ###3.40
118- ###2.54
119- ##-2.54
120- 1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
121- -1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
122- 
123- *** Output for integer type ***
124- 
125-  Input Integer numbers variation array is:
126- Array
127- (
128-     [0] => 0
129-     [1] => 1
130-     [2] => -1
131-     [3] => 2.7
132-     [4] => -2.7
133-     [5] => 23333333
134-     [6] => -23333333
135-     [7] => 1234
136- )
137- 
138- 
139- Integer Iteration 1
140- 0
141- 1
142- -1
143- 2
144- -2
145- 23333333
146- -23333333
147- 1234
148- 
149- Integer Iteration 2
150- 0
151- 1
152- -1
153- 2
154- -2
155- 23333333
156- -23333333
157- 1234
158- 
159- Integer Iteration 3
160- +0
161- +1
162- -1
163- +2
164- -2
165- +23333333
166- -23333333
167- +1234
168- 
169- Integer Iteration 4
170-       0
171-       1
172-      -1
173-       2
174-      -2
175- 23333333
176- -23333333
177-    1234
178- 
179- Integer Iteration 5
180- 0      
181- 1      
182- -1     
183- 2      
184- -2     
185- 23333333
186- -23333333
187- 1234   
188- 
189- Integer Iteration 6
190- 0000000
191- 0000001
192- -000001
193- 0000002
194- -000002
195- 23333333
196- -23333333
197- 0001234
198- 
199- Integer Iteration 7
200- 0      
201- 1      
202- -1     
203- 2      
204- -2     
205- 23333333
206- -23333333
207- 1234   
208- 
209- Integer Iteration 8
210- ######0
211- ######1
212- #####-1
213- ######2
214- #####-2
215- 23333333
216- -23333333
217- ###1234
218- 
219- *** Output for binary type ***
220- 
221-  Input  numbers variation array is:
222- Array
223- (
224-     [0] => 0
225-     [1] => 1
226-     [2] => -1
227-     [3] => 2.7
228-     [4] => -2.7
229-     [5] => 23333333
230-     [6] => -23333333
231-     [7] => 1234
232- )
233- 
234- 0
235- 1
236- 1111111111111111111111111111111111111111111111111111111111111111
237- 10
238- 1111111111111111111111111111111111111111111111111111111111111110
239- 1011001000000100111010101
240- 1111111111111111111111111111111111111110100110111111011000101011
241- 10011010010
242- 
243- *** Output for char type ***
244- 
245-  Input Characters variation array is:
246- Array
247- (
248-     [0] => a
249-     [1] => a
250-     [2] => 67
251-     [3] => -67
252-     [4] => 99
253- )
254- 
255- %0
256- %0
257- C
258- ½
259- c
260- 
261- *** Output for scientific type ***
262- 
263-  Input numbers variation array is:
264- Array
265- (
266-     [0] => 0
267-     [1] => 1
268-     [2] => -1
269-     [3] => 2.7
270-     [4] => -2.7
271-     [5] => 23333333
272-     [6] => -23333333
273-     [7] => 1234
274- )
275- 
276- 0.000000e+0
277- 1.000000e+0
278- -1.000000e+0
279- 2.700000e+0
280- -2.700000e+0
281- 2.333333e+7
282- -2.333333e+7
283- 1.234000e+3
284- 
285- *** Output for unsigned integer type ***
286- 
287-  Input Integer numbers variation array is:
288- Array
289- (
290-     [0] => 0
291-     [1] => 1
292-     [2] => -1
293-     [3] => 2.7
294-     [4] => -2.7
295-     [5] => 23333333
296-     [6] => -23333333
297-     [7] => 1234
298- )
299- 
300- 0
301- 1
302- 18446744073709551615
303- 2
304- 18446744073709551614
305- 23333333
306- 18446744073686218283
307- 1234
308- 
309- *** Output for octal type ***
310- 
311-  Input numbers variation array is:
312- Array
313- (
314-     [0] => 0
315-     [1] => 1
316-     [2] => -1
317-     [3] => 2.7
318-     [4] => -2.7
319-     [5] => 23333333
320-     [6] => -23333333
321-     [7] => 1234
322- )
323- 
324- 0
325- 1
326- 1777777777777777777777
327- 2
328- 1777777777777777777776
329- 131004725
330- 1777777777777646773053
331- 2322
332- 
333- *** Output for hexadecimal type ***
334- 
335-  Input numbers variation array is:
336- Array
337- (
338-     [0] => 0
339-     [1] => 1
340-     [2] => -1
341-     [3] => 2.7
342-     [4] => -2.7
343-     [5] => 23333333
344-     [6] => -23333333
345-     [7] => 1234
346- )
347- 
348- 0
349- 1
350- ffffffffffffffff
351- 2
352- fffffffffffffffe
353- 16409d5
354- fffffffffe9bf62b
355- 4d2
356- 
357- *** Output for string type ***
358- 
359-  Input Strings format variation array is:
360- Array
361- (
362-     [0] => %5s
363-     [1] => %-5s
364-     [2] => %r%%r05s
365-     [3] => %'#5s
366- )
367- 
368-  Input strings variation array is:
369- Array
370- (
371-     [0] => 
372-     [1] => abc
373-     [2] => aaa
374- )
375- 
376-      
377-   abc
378-   aaa
379-      
380- abc  
381- aaa  
382- 00000
383- 00abc
384- 00aaa
385- #####
386- ##abc
387- ##aaa
388- 
389- *** Output for '%g' type ***
390- 
391-  Input format variation array is:
392- Array
393- (
394-     [0] => %g
395-     [1] => %.0g
396-     [2] => %+g
397-     [3] => %-g
398-     [4] => %-1.2g
399-     [5] => %+1.2g
400-     [6] => %G
401-     [7] => %.0G
402-     [8] => %+G
403-     [9] => %-G
404-     [10] => %-1.2G
405-     [11] => %+1.2G
406- )
407- 
408- 123456
409- -123456
410- 1.0e+5
411- -1.0e+5
412- +123456
413- -123456
414- 123456
415- -123456
416- 1.2e+5
417- -1.2e+5
418- +1.2e+5
419- -1.2e+5
420- 123456
421- -123456
422- 1.0E+5
423- -1.0E+5
424- +123456
425- -123456
426- 123456
427- -123456
428- 1.2E+5
429- -1.2E+5
430- +1.2E+5
431- -1.2E+5
432- 
433- *** Output for '%%%.2f' as the format parameter ***
434- %12345678900.00
435- 
436- *** Output for '%%' as the format parameter ***
437- %
438- 
439- *** Output for precision value more than maximum ***
440- 
441- Notice: printf(): Requested precision of 988 digits was truncated to PHP maximum of %d digits in %s on line %d
442- 12345678900.0000000000%d
443- 
444- *** Output for invalid width(-15) specifier ***
445- Unknown format specifier "-"
446- 
447- *** Output for '%F' as the format parameter ***
448- 12345678900.000000
449- 
450- *** Output for '%X' as the format parameter ***
451- C
452- 
453- *** Output  with no format parameter ***
454- 12345
455- 
456- *** Output for multiple format parameters ***
457- 12345  abcdefghjklmnpqrstuvwxyz  12345
458- 
459- 
460- *** Output for excess of mixed type arguments  ***
461- abcdefghjklmnpqrstuvwxyz
462- 
463- *** Output for string format parameter and integer type argument ***
464- 12345
465- 
466- *** Output for integer format parameter and string type argument ***
467- 0
008+ Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
009+ Stack trace:
010+ #0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('%d dingy%sflem%...', NULL)
011+ #1 /Volumes/WebServer/Projects/php.fprint/tests/files/printf_64bit.php(32): user_printf('%d dingy%sflem%...', 5, 'voudras')
012+ #2 {main}
013+   thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22
