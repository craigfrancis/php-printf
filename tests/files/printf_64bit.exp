*** Output for zero argument ***
printf() expects at least 1 argument, 0 given

*** Output for insufficient number of arguments ***
Error found: 5 arguments are required, 3 given
*** Output for scalar argument ***
3

*** Output for float type ***

 Input Float numbers variation array is:
Array
(
    [0] => 0
    [1] => 1
    [2] => -1
    [3] => 0.32
    [4] => -0.32
    [5] => 3.4-3.4
    [6] => 2.54
    [7] => -2.54
    [8] => 1.2345678E+99
    [9] => -1.2345678E+99
)


Float Iteration 1
0.000000
1.000000
-1.000000
0.320000
-0.320000
3.400000
2.540000
-2.540000
1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000

Float Iteration 2
0.000000
1.000000
-1.000000
0.320000
-0.320000
3.400000
2.540000
-2.540000
1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000

Float Iteration 3
+0.000000
+1.000000
-1.000000
+0.320000
-0.320000
+3.400000
+2.540000
-2.540000
+1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.000000

Float Iteration 4
   0.00
   1.00
  -1.00
   0.32
  -0.32
   3.40
   2.54
  -2.54
1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00

Float Iteration 5
0.00   
1.00   
-1.00  
0.32   
-0.32  
3.40   
2.54   
-2.54  
1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00

Float Iteration 6
0000.00
0001.00
-001.00
0000.32
-000.32
0003.40
0002.54
-002.54
1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00

Float Iteration 7
0.00000
1.00000
-1.0000
0.32000
-0.3200
3.40000
2.54000
-2.5400
1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00

Float Iteration 8
###0.00
###1.00
##-1.00
###0.32
##-0.32
###3.40
###2.54
##-2.54
1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00
-1234567799999999974395005011934353453808157274826416694779123657996479008398428960266851879135215616.00

*** Output for integer type ***

 Input Integer numbers variation array is:
Array
(
    [0] => 0
    [1] => 1
    [2] => -1
    [3] => 2.7
    [4] => -2.7
    [5] => 23333333
    [6] => -23333333
    [7] => 1234
)


Integer Iteration 1
0
1
-1
2
-2
23333333
-23333333
1234

Integer Iteration 2
0
1
-1
2
-2
23333333
-23333333
1234

Integer Iteration 3
+0
+1
-1
+2
-2
+23333333
-23333333
+1234

Integer Iteration 4
      0
      1
     -1
      2
     -2
23333333
-23333333
   1234

Integer Iteration 5
0      
1      
-1     
2      
-2     
23333333
-23333333
1234   

Integer Iteration 6
0000000
0000001
-000001
0000002
-000002
23333333
-23333333
0001234

Integer Iteration 7
0      
1      
-1     
2      
-2     
23333333
-23333333
1234   

Integer Iteration 8
######0
######1
#####-1
######2
#####-2
23333333
-23333333
###1234

*** Output for binary type ***

 Input  numbers variation array is:
Array
(
    [0] => 0
    [1] => 1
    [2] => -1
    [3] => 2.7
    [4] => -2.7
    [5] => 23333333
    [6] => -23333333
    [7] => 1234
)

0
1
1111111111111111111111111111111111111111111111111111111111111111
10
1111111111111111111111111111111111111111111111111111111111111110
1011001000000100111010101
1111111111111111111111111111111111111110100110111111011000101011
10011010010

*** Output for char type ***

 Input Characters variation array is:
Array
(
    [0] => a
    [1] => a
    [2] => 67
    [3] => -67
    [4] => 99
)

%0
%0
C
½
c

*** Output for scientific type ***

 Input numbers variation array is:
Array
(
    [0] => 0
    [1] => 1
    [2] => -1
    [3] => 2.7
    [4] => -2.7
    [5] => 23333333
    [6] => -23333333
    [7] => 1234
)

0.000000e+0
1.000000e+0
-1.000000e+0
2.700000e+0
-2.700000e+0
2.333333e+7
-2.333333e+7
1.234000e+3

*** Output for unsigned integer type ***

 Input Integer numbers variation array is:
Array
(
    [0] => 0
    [1] => 1
    [2] => -1
    [3] => 2.7
    [4] => -2.7
    [5] => 23333333
    [6] => -23333333
    [7] => 1234
)

0
1
18446744073709551615
2
18446744073709551614
23333333
18446744073686218283
1234

*** Output for octal type ***

 Input numbers variation array is:
Array
(
    [0] => 0
    [1] => 1
    [2] => -1
    [3] => 2.7
    [4] => -2.7
    [5] => 23333333
    [6] => -23333333
    [7] => 1234
)

0
1
1777777777777777777777
2
1777777777777777777776
131004725
1777777777777646773053
2322

*** Output for hexadecimal type ***

 Input numbers variation array is:
Array
(
    [0] => 0
    [1] => 1
    [2] => -1
    [3] => 2.7
    [4] => -2.7
    [5] => 23333333
    [6] => -23333333
    [7] => 1234
)

0
1
ffffffffffffffff
2
fffffffffffffffe
16409d5
fffffffffe9bf62b
4d2

*** Output for string type ***

 Input Strings format variation array is:
Array
(
    [0] => %5s
    [1] => %-5s
    [2] => %r%%r05s
    [3] => %'#5s
)

 Input strings variation array is:
Array
(
    [0] => 
    [1] => abc
    [2] => aaa
)

     
  abc
  aaa
     
abc  
aaa  
00000
00abc
00aaa
#####
##abc
##aaa

*** Output for '%g' type ***

 Input format variation array is:
Array
(
    [0] => %g
    [1] => %.0g
    [2] => %+g
    [3] => %-g
    [4] => %-1.2g
    [5] => %+1.2g
    [6] => %G
    [7] => %.0G
    [8] => %+G
    [9] => %-G
    [10] => %-1.2G
    [11] => %+1.2G
)

123456
-123456
1.0e+5
-1.0e+5
+123456
-123456
123456
-123456
1.2e+5
-1.2e+5
+1.2e+5
-1.2e+5
123456
-123456
1.0E+5
-1.0E+5
+123456
-123456
123456
-123456
1.2E+5
-1.2E+5
+1.2E+5
-1.2E+5

*** Output for '%%%.2f' as the format parameter ***
%12345678900.00

*** Output for '%%' as the format parameter ***
%

*** Output for precision value more than maximum ***

Notice: printf(): Requested precision of 988 digits was truncated to PHP maximum of %d digits in %s on line %d
12345678900.0000000000%d

*** Output for invalid width(-15) specifier ***
Unknown format specifier "-"

*** Output for '%F' as the format parameter ***
12345678900.000000

*** Output for '%X' as the format parameter ***
C

*** Output  with no format parameter ***
12345

*** Output for multiple format parameters ***
12345  abcdefghjklmnpqrstuvwxyz  12345


*** Output for excess of mixed type arguments  ***
abcdefghjklmnpqrstuvwxyz

*** Output for string format parameter and integer type argument ***
12345

*** Output for integer format parameter and string type argument ***
0