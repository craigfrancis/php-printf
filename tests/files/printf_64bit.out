*** Output for zero argument ***
Too few arguments to function user_printf(), 0 passed in /Volumes/WebServer/Projects/php.fprint/tests/files/printf_64bit.php on line 21 and exactly 1 expected

*** Output for insufficient number of arguments ***

Warning: Undefined variable $args in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29

Fatal error: Uncaught TypeError: user_vprintf(): Argument #2 ($values) must be of type array, null given, called in /Volumes/WebServer/Projects/php.fprint/functions.php on line 29 and defined in /Volumes/WebServer/Projects/php.fprint/functions.php:22
Stack trace:
#0 /Volumes/WebServer/Projects/php.fprint/functions.php(29): user_vprintf('%d dingy%sflem%...', NULL)
#1 /Volumes/WebServer/Projects/php.fprint/tests/files/printf_64bit.php(32): user_printf('%d dingy%sflem%...', 5, 'voudras')
#2 {main}
  thrown in /Volumes/WebServer/Projects/php.fprint/functions.php on line 22